### PR TITLE
Math and scenario

### PR DESCRIPTION
**Pull request by Krajiyah**

### CODE REVIEW

**Comment by tanaythan** [comment]
@Krajiyah What do you think about renaming to `PositionVector` from `Vector3`?

File: src/util/mod.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+pub mod math;
```
Comment (tanaythan) [inline src/util/mod.rs]: Nit: extra line here.

File: src/util/math.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+pub fn new_pos(src_pos: f64, v: f64, t: f64) -> f64 {
+    v * t + src_pos
+}
+
+fn sqr(x: f64) -> f64 {
+    x * x
+}
+
+pub fn detect_collide_sphere_to_sphere(x1: f64, y1: f64, x2: f64, y2: f64,
```
Comment (tanaythan) [inline src/util/math.rs]: Change this to use 2 `Vector3`s

File: src/util/math.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
+const COEFFICIENT_OF_FRICTION: f64 = 0.05;
```
Comment (tanaythan) [inline src/util/math.rs]: We should use `f32` instead as to avoid typecasting.

File: tests/test_scenario.rs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    let dt = 1.0;
+    
+    // Should not collide before starting updates
+    assert_eq!(false, sphere1.is_collided(sphere2));
+    
+    // After updates should collide
+    state.update_entities(dt);
+    state.update_entities(dt);
+    state.update_entities(dt);
+    assert_eq!(true, sphere1.is_collided(sphere2));
```
Comment (Krajiyah) [inline tests/test_scenario.rs]: @tanaythan This assertion fails...but seems the spheres are not changing after update_entities calls. I think this is most likely problem with Copy thingy in rust

File: src/util/math.rs (hunk 2 above)
Comment (Krajiyah) [reply src/util/math.rs]: Changed

File: src/util/math.rs (hunk 3 above)
Comment (Krajiyah) [reply src/util/math.rs]: Changed to `f32`

File: src/util/mod.rs (hunk 1 above)
Comment (Krajiyah) [reply src/util/mod.rs]: added

**Review by tanaythan — APPROVED** [review]

**Pull request merged**